src/structures/paging/page_table.rs

```diff
@@ -175,9 +191,38 @@ bitflags! {
         /// Can be only used when the no-execute page protection feature is enabled in the EFER
         /// register.
         const NO_EXECUTE =      1 << 63;
+
+        // Consider other as defined
+        #[cfg(feature = "dynamic_flags")]
+        const _ = !0;
+    }
+}
+
+
+#[cfg(feature = "dynamic_flags")]
+impl PageTableFlags {
+    pub const fn from_address_truncated(address: u64) -> PageTableFlags {
+        PageTableFlags::from_bits_retain(address & !PHYSICAL_ADDRESS_MASK)
+    }
+
+    pub const fn bits_dynamic(&self) -> u64 {
+        self.bits() & !PHYSICAL_ADDRESS_MASK
+    }
```

Comment:
I'd prefer it if we just had `PageTableFlags::bits`. If we/users don't want to see address bits, they should mask them out when creating the `PageTableFlags` rather than masking them later.